src/python/bot/tasks/train_gradientfuzz_task.py

```diff
@@ -13,9 +13,251 @@
+      str(num_epochs), run_constants.BATCH_SIZE_FLAG,
+      str(batch_size), run_constants.VAL_BATCH_SIZE_FLAG,
+      str(val_batch_size), run_constants.ARCHITECTURE_FLAG,
+      constants.NEUZZ_ONE_HIDDEN_LAYER_MODEL
+  ]
+
+  logs.log('Launching training with the following arguments: "{}".'.format(
+      str(args_list)))
+
+  # Run process in gradientfuzz directory.
+  gradientfuzz_trainer = new_process.ProcessRunner(sys.executable)
+
+  return gradientfuzz_trainer.run_and_wait(
+      args_list,
+      cwd=GRADIENTFUZZ_SCRIPTS_DIR,
+      timeout=run_constants.TRAIN_TIMEOUT), run_name
+
+
+def get_model_dir(run_name):
+  """
+  Gets full path to model directory.
+
+  Args:
+      run_name (str): Self-explanatory.
+
+  Returns:
+      (str): Full path to model dir.
+  """
+  for full_path in glob.glob(
+      os.path.join(GRADIENTFUZZ_SCRIPTS_DIR, constants.MODEL_DIR, '*', '*')):
+    if os.path.basename(full_path) == run_name:
+      return full_path
+  return None
 
 
 def execute_task(fuzzer_name, job_type):
-  logs.log('Got a gradientfuzzer task with fuzzer name: {} and job type: {}.'
-           .format(fuzzer_name, job_type))
+  """
+  Performs GradientFuzz model training.
+
+  Grabs input corpus and processes it for inputs/labels.
+  Then trains a NEUZZ-like model and uploads model dir
+  (with all weights and metadata) to GCS.
+
+  Args:
+    fuzzer_name (str): Name of fuzzer, e.g. libpng_read_fuzzer.
+    job_type (str): Job type, e.g. libfuzzer_chrome_asan.
+  """
+  if not job_type:
+    logs.log_error(
+        'job_type is not set when training GradientFuzz for fuzzer {}.'.format(
+            fuzzer_name))
+    return
+
+  # Sets up fuzzer binary build.
+  environment.set_value('FUZZ_TARGET', fuzzer_name)
+  build_manager.setup_build()
+  fuzzer_binary_path = environment.get_value('APP_PATH')
+
+  # Directory to place corpus. |FUZZ_INPUTS_DISK| is not size constrained.
+  temp_directory = environment.get_value('FUZZ_INPUTS_DISK')
+
+  # Recreates corpus dir without contents.
+  corpus_directory = get_corpus_directory(temp_directory, fuzzer_name)
+  shell.remove_directory(corpus_directory, recreate=True)
+
+  # This actually downloads corpus directory based on fuzzer name from GCS.
+  logs.log('Downloading corpus backup for {}.'.format(fuzzer_name))
+  if not ml_train_utils.get_corpus(corpus_directory, fuzzer_name):
+    logs.log_error(
+        'Failed to download corpus backup for {}.'.format(fuzzer_name))
+    return
+
+  # First, generate input/label pairs for training.
+  gen_inputs_labels_result, dataset_name = gen_inputs_labels(
+      corpus_directory, fuzzer_binary_path)
+
+  if gen_inputs_labels_result.timed_out:
+    logs.log_warn('Data gen script for {} timed out.'.format(fuzzer_name))
+
+  # Next, invoke training script.
+  num_inputs = len(glob.glob(os.path.join(corpus_directory, '*')))
+  testing = environment.get_value('GRADIENTFUZZ_TESTING') is not None
```

Comment:
nit: just bool() on the result of get_value.

Reply:
Ahhhhh fair point. I forgot that `bool(None)` yields `False` :)